src/lib/storage/run_length_segment/run_length_segment_iterable.hpp

```diff
@@ -136,8 +136,8 @@ class RunLengthSegmentIterable : public PointAccessibleSegmentIterable<RunLength
 
     explicit PointAccessIterator(const pmr_vector<T>* values, const pmr_vector<bool>* null_values,
                                  const pmr_vector<ChunkOffset>* end_positions,
-                                 const PosList::const_iterator position_filter_begin,
-                                 PosList::const_iterator position_filter_it)
+                                 const PosList::const_iterator&& position_filter_begin,
```

Comment:
I don't think `const T&&` ever makes any sense.

Reply:
I read a few SO threads about it and from my understanding, it's the same as a const parameter passed by value. It's not required by helps to state the intend that this is not being changed.

Reply:
But I have to admit that I did not really consider it in this PR. Would set it on appropriate parameters.
Your opinion on `const T&&`?

Reply:
> Your opinion on const T&&?

Should be illegal.